### ISSUE TITLE
Optionally add values for missing days on calendar conversion

### ISSUE THREAD

**Issue by Zeitsperre** [Ouranosinc/xclim#582]
Converting calendars from one that has more days e.g. `all_leap` or  `366_day` towards `default` will drop the appropriate number of days in order to remain consistent throughout the year, but the same is not true currently when performing the inverse.

In the course of some analyses, I realize that having the option to pad the year with an odd `np.NaN` when going from `noleap` to `proleptic_gregorian`/`default` calendar would be a useful option to have to ensure that days don't decouple when comparing multiple years between data of different data sources.

In other words: being able to ensure that 30 years (1981 to 2010) of `noleap` days goes from `len(days) = 11680` to `len(days) == 11688` in a default calendar.
